internal/tools/networkagent/main.go

```diff
@@ -0,0 +1,350 @@
+			}
+			// remove endpoints that are in the namespace as NICs
+			nsReq := &ncproxygrpc.DeleteNICRequest{
+				ContainerID:  req.ContainerID,
+				NicID:        nicID,
+				EndpointName: endpoint.Name,
+			}
+			if _, err := s.client.DeleteNIC(ctx, nsReq); err != nil {
+				log.G(ctx).WithField("name", endpoint.Name).Warn("failed to delete endpoint nic")
+			}
+			delete(s.endpointToNicID, endpoint.Name)
+		}
+	}
+	return &nodenetsvc.ConfigureNetworkingResponse{}, nil
+}
+
+func (s *service) ConfigureNetworking(ctx context.Context, req *nodenetsvc.ConfigureNetworkingRequest) (*nodenetsvc.ConfigureNetworkingResponse, error) {
+	log.G(ctx).WithField("req", req).Info("ConfigureNetworking request")
+
+	containerNamespaceID, ok := s.containerToNamespace[req.ContainerID]
+	if !ok {
+		return nil, fmt.Errorf("no namespace was previously created for containerID %s", req.ContainerID)
+	}
+
+	if req.RequestType == nodenetsvc.RequestType_Setup {
+		return s.addHelper(ctx, req, containerNamespaceID)
+	}
+	return s.teardownHelper(ctx, req, containerNamespaceID)
+}
+
+func (s *service) GetHostLocalIpAddress(ctx context.Context, req *nodenetsvc.GetHostLocalIpAddressRequest) (*nodenetsvc.GetHostLocalIpAddressResponse, error) {
+	return &nodenetsvc.GetHostLocalIpAddressResponse{IpAddr: ""}, nil
+}
+
+func (s *service) PingNodeNetworkService(ctx context.Context, req *nodenetsvc.PingNodeNetworkServiceRequest) (*nodenetsvc.PingNodeNetworkServiceResponse, error) {
+	return &nodenetsvc.PingNodeNetworkServiceResponse{}, nil
+}
+
+func main() {
+	var err error
+	ctx := context.Background()
+
+	flag.Parse()
+	conf, err := readConfig(*configPath)
+	if err != nil {
+		log.G(ctx).WithError(err).Fatalf("failed to read network agent's config file at %s", *configPath)
+		os.Exit(1)
+	}
+	log.G(ctx).WithFields(logrus.Fields{
+		"config path": *configPath,
+		"conf":        conf,
+	}).Info("network agent configuration")
+
+	sigChan := make(chan os.Signal, 1)
+	serveErr := make(chan error, 1)
+	defer close(serveErr)
+	signal.Notify(sigChan, syscall.SIGINT)
+	defer signal.Stop(sigChan)
+
+	grpcClient, err := grpc.Dial(
+		conf.GRPCAddr,
+		grpc.WithInsecure(),
+	)
+	if err != nil {
+		log.G(ctx).WithError(err).Errorf("failed to connect to ncproxy at %s", conf.GRPCAddr)
+		os.Exit(1)
+	}
+	defer grpcClient.Close()
+
+	log.G(ctx).WithField("addr", conf.GRPCAddr).Info("connected to ncproxy")
+	ncproxyClient := ncproxygrpc.NewNetworkConfigProxyClient(grpcClient)
+	service := &service{
+		conf:                 conf,
+		client:               ncproxyClient,
+		containerToNamespace: make(map[string]string),
+		endpointToNicID:      make(map[string]string),
+		containerToNetwork:   make(map[string]string),
+	}
+	server := grpc.NewServer()
+	nodenetsvc.RegisterNodeNetworkServiceServer(server, service)
+
+	grpcListener, err := net.Listen("tcp", conf.NodeNetSvcAddr)
+	if err != nil {
+		log.G(ctx).WithError(err).Errorf("failed to listen on %s", grpcListener.Addr().String())
+		os.Exit(1)
```

Comment:
Change to Fatalf https://github.com/microsoft/hcsshim/pull/1067/files#r712853897